src/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs

```diff
@@ -0,0 +1,621 @@
+            + "\\s*=\\s*"                                               // the equal sign divides the key and value parts
+            + "(?<value>"
+            + "(\\{([^\\}\u0000]|\\}\\})*\\})"                         // quoted string, starts with { and ends with }
+            + "|"
+            + "((?![\\{\\s])"                                          // unquoted value must not start with { or space, would also like = but too late to change
+            + "([^;\\s\\p{Cc}]|\\s+[^;\\s\\p{Cc}])*"                  // control characters must be quoted
+
+            + ")" // VSTFDEVDIV 94761: although the spec does not allow {}
+                  // embedded within a value, the retail code does.
+                  // +  "(?<![\\}]))"                                            // unquoted value must not stop with }
+
+            + ")(\\s*)(;|[\u0000\\s]*$)"                               // whitespace after value up to semicolon or end-of-line
+            + ")*"                                                      // repeat the key-value pair
+            + "[\\s;]*[\u0000\\s]*"                                     // traling whitespace/semicolons (DataSourceLocator), embedded nulls are allowed only in the end
+        ;
+
+        private static readonly Regex s_connectionStringRegex = new Regex(ConnectionStringPattern, RegexOptions.ExplicitCapture);
+#endif
+        private const string ConnectionStringValidKeyPattern = "^(?![;\\s])[^\\p{Cc}]+(?<!\\s)$"; // key not allowed to start with semi-colon or space or contain non-visible characters or end with space
+        private const string ConnectionStringValidValuePattern = "^[^\u0000]*$";                    // value not allowed to contain embedded null
+        private const string ConnectionStringQuoteValuePattern = "^[^\"'=;\\s\\p{Cc}]*$";           // generally do not quote the value if it matches the pattern
+        private const string ConnectionStringQuoteOdbcValuePattern = "^\\{([^\\}\u0000]|\\}\\})*\\}$"; // do not quote odbc value if it matches this pattern
+        internal const string DataDirectory = "|datadirectory|";
+
+        private static readonly Regex s_connectionStringValidKeyRegex = new Regex(ConnectionStringValidKeyPattern);
+        private static readonly Regex s_connectionStringValidValueRegex = new Regex(ConnectionStringValidValuePattern);
+        private static readonly Regex s_connectionStringQuoteValueRegex = new Regex(ConnectionStringQuoteValuePattern);
+
+        // connection string common keywords
+        private static class KEY
+        {
+            internal const string Integrated_Security = "integrated security";
+            internal const string Password = "password";
+            internal const string Persist_Security_Info = "persist security info";
+            internal const string User_ID = "user id";
+        };
+
+        // known connection string common synonyms
+        private static class SYNONYM
```

Comment:
Do these types need to be cased this way, or could they just be Key and Synonym?